lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -0,0 +1,531 @@
+           have any sensible weights."""
+        nbhood_data = np.empty(self.weights_cube.data.shape)
+        nbhood_data[:] = np.nan
+        nbhooded_cube = self.weights_cube.copy(nbhood_data)
+        message = "Sum of weights must be > 0.0"
+        with self.assertRaisesRegexp(ValueError, message):
+            self.plugin.reweight_weights(nbhooded_cube)
+
+    def test_no_NaNs_in_nbhooded_cube_and_masked_weights(self):
+        """No NaNs in the neighbourhood cube, but masked weights."""
+        nbhooded_cube = self.weights_cube.copy()
+
+        self.weights_cube.data = np.ma.masked_array(self.weights_cube.data,
+                                                    mask=self.mask)
+        expected_weights = self.weights_cube.data.copy()
+        self.plugin.reweight_weights(nbhooded_cube)
+        self.assertArrayAlmostEqual(expected_weights.data,
+                                    self.weights_cube.data.data)
+        self.assertArrayAlmostEqual(expected_weights.mask,
+                                    self.weights_cube.data.mask)
+
+    def test_some_NaNs_in_nbhooded_cube(self):
+        """Some NaNs in the neighbourhood cube, so reweighting is needed"""
+        nbhood_data = np.ones((3, 5, 5))
+        nbhood_data[0, 0:2, 0] = np.nan
+        nbhood_data[2, 2:4, 4] = np.nan
+
+        expected_weights = np.array([[[0.0, 0.7, 0.0, 0.0, 0.0],
+                                      [0.0, 0.3, 0.0, 0.0, 0.0],
+                                      [0.3, 0.1, 0.0, 0.0, 0.0],
+                                      [0.0, 0.0, 0.0, 0.0, 0.0],
+                                      [0.0, 0.0, 0.0, 0.0, 0.0]],
+                                     [[1.0, 0.3, 1.0, 1.0, 1.0],
+                                      [1.0, 0.7, 1.0, 1.0, 1.0],
+                                      [0.7, 0.9, 1.0, 1.0, 1.0],
+                                      [1.0, 1.0, 1.0, 0.9, 1.0],
+                                      [1.0, 1.0, 1.0, 0.6, 0.2]],
+                                     [[0.0, 0.0, 0.0, 0.0, 0.0],
+                                      [0.0, 0.0, 0.0, 0.0, 0.0],
+                                      [0.0, 0.0, 0.0, 0.0, 0.0],
+                                      [0.0, 0.0, 0.0, 0.1, 0.0],
+                                      [0.0, 0.0, 0.0, 0.4, 0.8]]])
+        nbhooded_cube = self.weights_cube.copy(nbhood_data)
+        self.plugin.reweight_weights(nbhooded_cube)
+        self.assertArrayAlmostEqual(expected_weights, self.weights_cube.data)
+
+    def test_some_NaNs_in_nbhooded_cube_and_masked_weights(self):
+        """Some NaNs in the neighbourhood cube, so reweighting is needed.
+           As the points with NaNs in the neighbourhood cube are masked in the
+           weights cube then they are not reweighted."""
+        nbhood_data = np.ones((3, 5, 5))
+        nbhood_data[0, 0:2, 0] = np.nan
+        nbhood_data[2, 2:4, 4] = np.nan
+        nbhooded_cube = self.weights_cube.copy()
```

Comment:
This definition of the neighbourhooded cube comes up several times in the same test class.  Suggest moving into setUp() and redefining only where the data matrix is different (nbhood_cube = self.nbhood_cube.copy(new_data)).

Reply:
Had quite a bit of a tidy up, let me know if you think it's enough.